addon/-private/formatters/format-date.ts

```diff
@@ -21,33 +21,33 @@ const DATE_TIME_OPTIONS = [
   'second',
   'timeZoneName',
   'hourCycle',
-];
+] as readonly (keyof Intl.DateTimeFormatOptions)[];
 
 /**
  * @private
  * @hide
  */
-export default class FormatDate extends Formatter {
-  static type = 'date';
+export default class FormatDate extends Formatter<Intl.DateTimeFormatOptions> {
+  static readonly type = 'date';
 
-  constructor(config) {
-    super(config);
-
-    this.createNativeFormatter = memoize((locales, options) => {
-      return new Intl.DateTimeFormat(locales, options);
-    });
-  }
+  createNativeFormatter = memoize((locales: string | string[], options: Intl.DateTimeFormatOptions) => {
+    return new Intl.DateTimeFormat(locales, options);
+  });
 
   get options() {
     return DATE_TIME_OPTIONS;
   }
 
-  format(locale, value, formatOptions) {
+  format(
+    locale: string | string[],
+    value: ConstructorParameters<typeof Date>[0],
+    formatOptions: Intl.DateTimeFormatOptions & BaseOptions
+  ): string {
     const formatterOptions = this.readOptions(formatOptions);
 
     this.validateFormatterOptions(locale, formatterOptions);
     const formatterInstance = this.createNativeFormatter(locale, formatterOptions);
 
-    return formatterInstance.format(new Date(value), formatOptions);
+    return formatterInstance.format(new Date(value));
```

Comment:
According to MDN, there is no second parameter:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/format#Syntax